File: apps/frontend/features/summaries/lib/src/domain/entities/reader_summary_narrative_section.dart
enum ReaderSummaryNarrativeSectionKind {
  lead,
  mainSignal,
  whyItMatters,
  secondarySignal,
  watch,
}

final class ReaderSummaryNarrativeSection {
  const ReaderSummaryNarrativeSection({
    required this.id,
    required this.kind,
    required this.title,
    required this.text,
    required this.citationIds,
    this.storyClusterId,
  });

  final String id;
  final ReaderSummaryNarrativeSectionKind kind;
  final String title;
  final String text;
  final List<String> citationIds;
  final String? storyClusterId;
}

File: libs/summary/domain/entities/reader-summary-narrative-section.ts
export const readerSummaryNarrativeSectionKinds = [
  "lead",
  "main_signal",
  "why_it_matters",
  "secondary_signal",
  "watch",
] as const;

export type ReaderSummaryNarrativeSectionKind =
  (typeof readerSummaryNarrativeSectionKinds)[number];

export type ReaderSummaryNarrativeSection = {
  readonly id: string;
  readonly kind: ReaderSummaryNarrativeSectionKind;
  readonly title: string;
  readonly text: string;
  readonly citationIds: readonly string[];
  readonly storyClusterId?: string;
};
